test/jdk/sun/security/krb5/auto/HttpsCB.java

```diff
@@ -1,311 +1,311 @@
+        // Write a customized JAAS conf file, so that any kinit cache
+        // will be ignored.
+        System.setProperty("java.security.auth.login.config", OneKDC.JAAS_CONF);
+        File f = new File(OneKDC.JAAS_CONF);
+        FileOutputStream fos = new FileOutputStream(f);
+        fos.write((
+                "com.sun.security.jgss.krb5.initiate {\n" +
+                "    com.sun.security.auth.module.Krb5LoginModule required;\n};\n"
+                ).getBytes());
+        fos.close();
+
+        HttpServer h1 = httpd("Negotiate",
+                "HTTP/" + WEB_HOST + "@" + REALM_WEB, KRB5_TAB);
+        webPort = h1.getAddress().getPort();
+
+        cbtURL = new URL("https://" + WEB_HOST +":" + webPort + "/cbt");
+        normalURL = new URL("https://" + WEB_HOST +":" + webPort + "/normal");
+
+        java.net.Authenticator.setDefault(new java.net.Authenticator() {
+            public PasswordAuthentication getPasswordAuthentication () {
+                return new PasswordAuthentication(
+                        WEB_USER+"@"+REALM_WEB, WEB_PASS);
+            }
+        });
+
+        // Client-side SSLContext needs to ignore hostname mismatch
+        // and untrusted certificate.
+        SSLContext sc = SSLContext.getInstance("SSL");
+        sc.init(null, new TrustManager[] {
+                new X509ExtendedTrustManager() {
+                    public X509Certificate[] getAcceptedIssuers() {
+                        return null;
+                    }
+                    public void checkClientTrusted(X509Certificate[] chain,
+                            String authType, Socket socket) { }
+                    public void checkServerTrusted(X509Certificate[] chain,
+                            String authType, Socket socket) { }
+                    public void checkClientTrusted(X509Certificate[] chain,
+                            String authType, SSLEngine engine) { }
+                    public void checkServerTrusted(X509Certificate[] chain,
+                            String authType, SSLEngine engine) { }
+                    public void checkClientTrusted(X509Certificate[] certs,
+                            String authType) { }
+                    public void checkServerTrusted(X509Certificate[] certs,
+                            String authType) { }
+                }
+        }, null);
+
+        Asserts.assertEQ(visit(sc, cbtURL), expected1);
+        Asserts.assertEQ(visit(sc, normalURL), expected2);
+    }
+
+    static boolean visit(SSLContext sc, URL url) {
+        try {
+            HttpsURLConnection conn = (HttpsURLConnection)
+                    url.openConnection(Proxy.NO_PROXY);
+            conn.setSSLSocketFactory(sc.getSocketFactory());
+            BufferedReader reader;
+            reader = new BufferedReader(new InputStreamReader(
+                    conn.getInputStream()));
+            return reader.readLine().equals(CONTENT);
+        } catch (Exception e) {
+            return false;
```

Comment:
Should we log that we have received the excepted exception here? Shouldn't the catch clause only list the exceptions that we are expecting?

Reply:
It's `java.net.SocketException: Unexpected end of file from server`. Does not include any CBT words so don't know if it's worth parsing.

Reply:
Thanks. Then it would be better to catch only `SocketException` here rather than `Exception`.

Reply:
I'll make it catch `IOException`